src/ucp/api/ucp.h

```diff
@@ -631,6 +633,120 @@ void ucp_worker_progress(ucp_worker_h worker);
 
 
 /**
+ * @ingroup UCP_WAKEUP
+ * @brief Obtain an event file descriptor for event notification.
+ *
+ * This routine returns a valid file descriptor for polling functions.
+ * The file descriptor will get signaled when an event occurs, as part of the
+ * wake-up mechanism. Signaling means a call to poll() or select() with this
+ * file descriptor will return at this point, with this descriptor marked as the
+ * reason (or one of the resons) the function has returned.
+ *
+ * The worker must be armed before waiting on an event (must be re-armed after
+ * it has been signaled for re-use) with @ref ucp_worker_arm.
+ * The events triggering a signal of the file descriptor depend on the
+ * interfaces used by the worker and are defined in the transport layer, and
+ * typically represent a request completion or newly available resources.
+ * The file descriptor should be closed by invoking close().
+ *
+ * The wake-up mechanism exists to allow for the user process to register for
+ * notifications on events of the underlying interfaces, and wait until such
+ * occur. This is an alternative to repeated polling for request completion.
+ * The goal is to allow for waiting while consuming minimal resources from the
+ * system. This is recommended for cases where traffic is infrequent, and
+ * latency can be traded for lower resource consumption while waiting for it.
+ *
+ * There are two alternative ways to use the wakeup mechanism: the first is the
+ * file descriptor obtained per worker (this function) and the second is the
+ * @ref ucp_worker_wait function for waiting on the next event internally.
+ *
+ * @param [in]  worker    Worker of notified events.
+ * @param [out] fd        File descriptor.
+ */
+ucs_status_t ucp_worker_get_efd(ucp_worker_h worker, int *fd_p);
+
+
+/**
+ * @ingroup UCP_WAKEUP
+ * @brief Wait for an event of the worker.
+ *
+ * This routine waits (blocking) until an event has happened, as part of the
+ * wake-up mechanism.
+ *
+ * The worker must be armed before waiting on an event (must be re-armed after
```

Comment:
I don't think we have to repeat this paragraph in every function. I think the arm description has to go to the arm function and you don't have to repeat it everywhere